### ISSUE TITLE
Add helper variable progress to campaign

### ISSUE THREAD

**Issue by angaither** [DoSomethingArchive/legacy-website#4282]
this field displays the total progress of the number of x things y'ed

if left blank, use the total number of approved reportback quanity goes there
otherwise, use the number in the new total progress field

once a new reportback is approved, add to the progress value set in the campaign node (if it's overridden we are adding to that overridden value)

**Comment by aaronschachter** [comment]
@angaither FYI that we're using `dosomething_helper_variables` to store the current Reportback count by status of each Campaign and Cause's. This issue is also really similar to #4051 which is an open issue to store current count of Signups, which suggests also using the helper variables or @mshmsh5000 mentions storing it in Redis in that thread.

Fields could work here, it would just be a different implementation from the current Reportback Status counts. I used the helper variables specifically for the Reportback stuff to avoid `node_save` upon each Reportback File create/update, and to avoid adding 4 fields to the Campaign node / Cause vocab.

**Comment by aaronschachter** [comment]
This is also perhaps a duplicate of #3899 ?

**Comment by angaither** [comment]
Ah, yes. it does look like a dup of #3899, however this need to be able to potentially be override, like the totals in a campaign run.

**Comment by aaronschachter** [comment]
My $0.02 is that we should keep the build simpler by simply not allowing overrides.

If that's not an option, we could always build an edit form for the Helper variables, to avoid storing all this stuff in fields.  Keeping in fields has its pros and cons but I think the biggest con would be a frequent `node_save`.

**Comment by angaither** [comment]
The case for overrides comes from campaigns with external partners, where we hear count numbers higher than than what we have in the rb_quanitity, like H&M reports we got 30K more pounds of clothes, but only 10k was reported back to us.

**Comment by aaronschachter** [comment]
Ah, I see. Thanks for clarifying / documenting!  Those are the sort of scenarios where it can be easy to forget why we built something a certain way when we look back at it in the future :watch:

**Comment by angaither** [comment]
Closed with #4320
